### ISSUE TITLE
deprecation and future warnings with matplotlib

### ISSUE THREAD

**Issue by moonshoes87** [PmagPy/PmagPy#444]
**What command was run:**

pmag_gui.py -h

**What happened:**

    /Users/nebula/Python/PmagPy/programs/pmag_gui.py:11: UserWarning: matplotlib.pyplot as already been imported, this call will have no effect.
      matplotlib.use('WXAgg')

**What I expected to happen:**

Backend should have been set to WXAgg, instead of being stuck as MacOSX.  Problem is solved by downgrading to 2.3.3.  Possibly this is a new matplotlib bug?  It is of concern if new users download matplotlib 3, though.

**Data file(s):**

**Screenshot(s):**

**PmagPy version:**

**Comment by moonshoes87** [comment]
https://github.com/matplotlib/matplotlib/issues/12362

**Comment by moonshoes87** [comment]
This problem is present in 3.0.1, but should be solved in 3.0.2 (see matplotlib issue referenced above).

**Comment by njarboe** [comment]
I know you told me somewhere how to downgrade, but I can't find the info at the moment.
conda install matplotlib=2.3.3 does not work as it looks like anaconda has removed it from their repos.

**Comment by moonshoes87** [comment]
It looks like this won't be resolved until 3.1.

**Comment by moonshoes87** [comment]
Relevant issue has been closed in matplotlib, fix should be available in 3.1: https://github.com/matplotlib/matplotlib/issues/13096

**Comment by moonshoes87** [comment]
All tests work except for pmagpy_tests.test_pmagplotlib.TestPlotMagMap.test_success in the Linux test environment.  This test does work on my Linux VM, though.  Possibly the version of wx is the problem?

**Comment by moonshoes87** [comment]
Nope, having wxPython 4.0.4 instead of 4.0.6 does NOT fix the above error message :(

**Comment by moonshoes87** [comment]
So, in general matplotlib 3.1 works just fine, for Windows, MacOS, and Linux.  It fails only in one case in the test environment.  I haven't yet updated the Cookbook instructions to reflect this, but PmagPy is currently compatible with either matplotlib 2.2.3 or 3.1.  Yay!

**Comment by moonshoes87** [comment]
3.1.0 works, but some issues with 3.1.2 need to be fixed:

<img width="982" alt="Screen Shot 2020-01-28 at 2 39 21 PM" src="https://user-images.githubusercontent.com/1836578/73311725-09212880-41dc-11ea-8e66-86acccc6ed42.png">

**Comment by moonshoes87** [comment]
So, it looks like 3.0.3 and only 3.0.3 will work without crashing on Linux.  We're going to need to address the deprecation warnings and the future warnings.  3.0.3 and 3.1.0 work on OSX.

**Comment by moonshoes87** [comment]
Further testing -- 3.0.3 and 3.1.0 both work on linux.  It's just the testing environment that is broken.  So I'm going to leave this alone for now.  We do still need to get those deprecation warnings and future warnings, though.
